src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs

```diff
@@ -153,6 +153,13 @@ private bool TryGetTextForSymbol(SyntaxToken token, SemanticModel semanticModel,
                 symbol = ((ILocalSymbol)symbol).Type;
             }
 
+            // Range variable: use the type
+            if (symbol is IRangeVariableSymbol)
+            {
+                var info = semanticModel.GetSpeculativeTypeInfo(token.SpanStart, SyntaxFactory.ParseName(symbol.Name), SpeculativeBindingOption.BindAsExpression);
+                symbol = info.Type;
```

Comment:
1. There's no way to just ask a Range variable what it's type is?
2. If there's no way, that doesn't mean we should try to invent a way by getting the name of the range variable, parsing it, and hoping it speculatively binds.  This is very fragile.
3. Smelly code elsewhere isn't a reason to make more code smellier :)
4. If we really do want this (and I can see at least 4 places in the code where we try to get this), then we should introduce a real API on SemanticModel (or IRangeVariable) to just give us this info.